lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/api/VortexUserCode.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.vortex.api;
+
+/**
+ * The starting point of Vortex user thread.
+ */
+public interface VortexUserCode {
+  /**
+   * Implement this method, using VortexThreadPool to run jobs on Vortex.
+   * Your implementation of this interface will be instantiated by Tang
+   * and this method will be called upon the start of REEF Driver.
+   */
+  void start(final VortexThreadPool vortexThreadPool);
```

Comment:
As per the conference call today, we'll stick with the current approach. Let's revisit this later when we have more concrete Vortex applications.